webapp/tests/karma/ts/services/create-user-for-contacts.service.spec.ts

```diff
@@ -0,0 +1,373 @@
+        { provide: SessionService, useValue: sessionService },
+      ]
+    });
+
+    service = TestBed.inject(CreateUserForContactsService);
+  });
+
+  afterEach(() => sinon.restore());
+
+  const assertContactNotUpdated = () => {
+    expect(medicDb.put.callCount).to.equal(0);
+    expect(dbSyncService.isSyncInProgress.callCount).to.equal(0);
+    expect(dbSyncService.sync.callCount).to.equal(0);
+    expect(sessionService.logout.callCount).to.equal(0);
+  };
+
+  describe('getUserContact', () => {
+    it('returns the user contact', async () => {
+      userSettingsService.get.resolves({ contact_id: ORIGINAL_CONTACT._id });
+      medicDb.get.withArgs(ORIGINAL_CONTACT._id).resolves(ORIGINAL_CONTACT);
+
+      const userContact = await service.getUserContact();
+
+      expect(userContact).to.equal(ORIGINAL_CONTACT);
+      expect(userSettingsService.get.callCount).to.equal(1);
+      expect(dbService.get.callCount).to.equal(1);
+      expect(medicDb.get.callCount).to.equal(1);
+      expect(medicDb.get.args[0]).to.deep.equal([ORIGINAL_CONTACT._id]);
+    });
+
+    it('returns undefined if the user has no contact id', async () => {
+      userSettingsService.get.resolves({ });
+
+      const userContact = await service.getUserContact();
+
+      expect(userContact).to.be.undefined;
+      expect(userSettingsService.get.callCount).to.equal(1);
+      expect(dbService.get.callCount).to.equal(0);
+      expect(medicDb.get.callCount).to.equal(0);
+    });
+
+    it(`returns undefined if the user's contact id is not associated with any document`, async () => {
+      userSettingsService.get.resolves({ contact_id: ORIGINAL_CONTACT._id });
+      medicDb.get.withArgs(ORIGINAL_CONTACT._id).rejects({ status: 404 });
+
+      const userContact = await service.getUserContact();
+
+      expect(userContact).to.be.undefined;
+      expect(userSettingsService.get.callCount).to.equal(1);
+      expect(dbService.get.callCount).to.equal(1);
+      expect(medicDb.get.callCount).to.equal(1);
+      expect(medicDb.get.args[0]).to.deep.equal([ORIGINAL_CONTACT._id]);
+    });
+
+    it(`throws an error is an error is encountered getting the user's contact`, async () => {
+      userSettingsService.get.resolves({ contact_id: ORIGINAL_CONTACT._id });
+      medicDb.get.withArgs(ORIGINAL_CONTACT._id).rejects({ message: 'Server Error' });
+
+      try {
+        await service.getUserContact();
+        expect.fail('should have thrown an error');
+      } catch (err) {
+        expect(err.message).to.equal('Server Error');
+      }
+
+      expect(userSettingsService.get.callCount).to.equal(1);
+      expect(dbService.get.callCount).to.equal(1);
+      expect(medicDb.get.callCount).to.equal(1);
+      expect(medicDb.get.args[0]).to.deep.equal([ORIGINAL_CONTACT._id]);
+    });
+  });
+
+  describe('setReplaced', () => {
+    it('sets the given new contact and a status of PENDING when the user is offline', () => {
```

Comment:
Please be specific that these users have offline roles, not offline as in disconnected.